internal/destination/config.go

```diff
@@ -76,6 +76,10 @@ func (c *Config) parseFields(cfg map[string]string) error {
 			return fmt.Errorf("parse %q: %w", ConfigKeyRetryWait, err)
 		}
 
+		if retryWait <= 0 {
+			retryWait = defaultRetryWait
+		}
+
```

Comment:
Great catch! IMHO though, it's better to return an error. If a user provided we should either accept it as it is or return an error. Replacing it with something else would be unexpected, at least to me.

Reply:
I added some error returns for negative values and fix that 0 retries as a valid value.